ci/check-style.sh

```diff
@@ -52,26 +49,33 @@ find . \( -path ./.git -prune -o -path ./third_party -prune \
      -o \( -name '*.cc' -o -name '*.h' \) -print0 \
      | xargs -0 clang-format -i
 
+# Apply several transformations that cannot be enforced by clang-format:
+#     - Replace any #include for grpc++/* with grpcpp/*. The paths with grpc++
+#       are obsoleted by the gRPC team, so we should not use them in our code.
+#     - Replace grpc::<BLAH> with grpc::StatusCode::<BLAH>, the aliases in the
+#       `grpc::` namespace do not exist inside google.
+for file in $(find google/cloud -name '*.h' -o -name '*.cc' -print); do
+  # We used to run run `sed -i` to apply these changes, but that touches the
+  # files even if there are no changes applied, forcing a rebuild each time.
+  # So we first apply the change to a temporary file, and replace the original
+  # only if something changed.
+  sed -e 's/grpc::\([A-Z][A-Z_][A-Z_]*\)/grpc::StatusCode::\1/g' \
+      -e 's;#include <grpc\\+\\+/grpc\+\+.h>;#include <grpcpp/grpcpp.h>;' \
+      -e 's;#include <grpc\\+\\+/;#include <grpcpp/;' \
+      "${file}" > "${file}.tmp"
+  if cmp "${file}" "${file}.tmp"; then
+      rm -f "${file}.tmp"
+  else
+      mv "${file}.tmp" "${file}"
```

Comment:
Just a thought: do you need `mv -f` here? What if someone has `alias mv='mv -i'` in their shell?

FWIW, above you have `rm -f`, not simply `rm`...